march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -444,20 +449,18 @@ void MarchHardwareInterface::updateIMotionCubeState()
 
 void MarchHardwareInterface::iMotionCubeStateCheck(size_t joint_index)
 {
-  march::IMotionCubeState imc_state = march_robot_->getJoint(joint_names_[joint_index]).getIMotionCubeState();
-  if (imc_state.state == march::IMCState::FAULT)
-  {
-    this->march_robot_->stopEtherCAT();
-    std::ostringstream error_stream;
-    error_stream << "IMotionCube of joint " << joint_names_[joint_index].c_str() << " is in fault state "
-                 << imc_state.state.getString() << std::endl;
-    error_stream << "Detailed Error: " << imc_state.detailedErrorDescription << "(" << imc_state.detailedError << ")"
-                 << std::endl;
-    error_stream << "Motion Error: " << imc_state.motionErrorDescription << "(" << imc_state.motionError << ")"
-                 << std::endl;
-
-    throw std::runtime_error(error_stream.str());
-  }
+    march::IMotionCubeState imc_state = march_robot_->getJoint(joint_names_[joint_index]).getIMotionCubeState();
+    if (imc_state.state == march::IMCState::FAULT)
+    {
+        ROS_ERROR(<< "IMotionCube of joint " << joint_names_[joint_index].c_str() << " is in fault state "
+                  << imc_state.state.getString() << std::endl );
+        ROS_ERROR(<< "Detailed Error: " << imc_state.detailedErrorDescription << "(" << imc_state.detailedError << ")"
+                  << std::endl);
+        ROS_ERROR(<< "Motion Error: " << imc_state.motionErrorDescription << "(" << imc_state.motionError << ")"
+                  << std::endl;)
+        fault_state = true;
```

Comment:
```suggestion
        return false;
```
(it makes more sense to return false when something goes wrong and true when everything is okay)